server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -470,6 +468,39 @@ protected void updateIndicesReadOnly(Set<String> indicesToUpdate, ActionListener
             .execute(wrappedListener.map(r -> null));
     }
 
+    private synchronized void markIndicesAsReadOnly(Set<String> indices, ActionListener<Void> listener) {
+        if (indices.isEmpty() == false && diskThresholdSettings.isEnabled()) {
+            logger.trace("marking indices as read-only: [{}]", indices);
+            updateIndicesReadOnly(indices, listener, true);
+        } else {
+            listener.onResponse(null);
+        }
+    }
+
+    private synchronized void removeExistingIndexBlocksIfDisabled(boolean enabled) {
+        if (enabled) {
+            return;
+        }
+        ActionListener<Void> wrappedListener = ActionListener.wrap(
+            r -> {},
+            e -> logger.debug("removing read-only blocks from indices failed", e)
```

Comment:
Not sure, if we already have a simpler way or existing listeners to just log the failure.